cf/post-register-lambda.yaml

```diff
@@ -94,66 +142,97 @@ Resources:
-              experimentIds.push(experimentId)
+              experimentIds.push(experimentId);
               const role = item.role.S;
-              
+
               const addRoleParams = {
-                  TableName: userAccessTableName,
-                  Item: {
-                    userId,
-                    experimentId,
-                    projectId,
-                    role,
-                    createdDate: new Date().toISOString(),
-                  },
-                };
-            
-                const docClient = new AWS.DynamoDB.DocumentClient({
-                  region,
-                });
-            
-                return docClient.put(addRoleParams).promise();
-                
-            }))
-            
-            //remove rows from invite-access
-            await Promise.all(experimentIds.map((id)=>{
-                
+                TableName: userAccessTableName,
+                Item: {
+                  userId,
+                  experimentId,
+                  projectId,
+                  role,
+                  createdDate: new Date().toISOString(),
+                },
+              };
+
+              const docClient = new AWS.DynamoDB.DocumentClient({
+                region,
+              });
+
+              return docClient.put(addRoleParams).promise();
+            }));
+
+            console.log(`V1 :: Removing invitation record`)
+
+            // remove rows from invite-access
+            await Promise.all(experimentIds.map((id) => {
               const key = convertToDynamoDbRecord({
                 experimentId: id,
                 userEmail,
               });
-              
+
               const dynamoParams = {
                 TableName: inviteAccessTableName,
                 Key: key,
               };
 
               return db.deleteItem(dynamoParams).send();
-            }))
-            
+            }));
+          };
+
+          const addUserPermissionV2 = async (userEmail, userId) => {
+            const sns = new AWS.SNS({ region });
+
+            const message = {
+              userEmail,
+              userId
+            };
+
+            console.log(`V2 :: Sending SNS message for user ${message}`)
+
+            await sns.publish({
+              Message: JSON.stringify(message),
+              TopicArn: process.env.TOPIC_ARN,
+              MessageAttributes: {
+                type: {
+                  DataType: 'String',
+                  StringValue: 'PostUserRegistration'
+                }
+              }
+            }, () => {}).promise();
+          };
+
+          exports.handler = async (event, context, callback) => {
+
+            const {
+              email: userEmail,
+              sub: userId,
+            } = event.request.userAttributes;
+
+            console.log(`== Handling post-registration for ${userEmail} ==`)
+
+            await addUserPermissionV1(userEmail, userId);
```

Comment:
You can use Promise.all here for these calls

Reply:
I wrote it this way so that it can be easily removed once we move away from v1.